actors/builtin/miner/miner_actor.go

```diff
@@ -323,14 +326,20 @@ func (a Actor) ProveCommitSector(rt Runtime, params *ProveCommitSectorParams) *a
 			rt.Abort(exitcode.ErrIllegalState, "failed to delete precommit for sector %v: %v", sectorNo, err)
 		}
 
-		st.ProvingSet = st.ComputeProvingSet()
+		// Do not update proving set during challenge window
+		if !st.inChallengeWindow(rt) {
+			st.ProvingSet = st.Sectors
```

Comment:
haaaa this is why we do this here: note that with surprise this was unnecessary given the surprise was triggered by the powerActor, and so we could just update the provingSet there and nowhere else.